pkg/gofr/datasource/file/ftp/fs.go

```diff
@@ -0,0 +1,217 @@
+
+	f.conn = &Conn{conn}
+
+	// Login to FTP server
+	err = conn.Login(f.config.User, f.config.Password)
+	if err != nil {
+		log.Printf("Failed to login: %v", err)
+	} else {
+		log.Printf("Login Successful")
+	}
+}
+
+// Create creates an empty file on the FTP server.
+func (f *ftpFileSystem) Create(name string) (datasource.File, error) {
+	// empty io.Reader
+	emptyReader := new(bytes.Buffer)
+
+	if name == "" {
+		return nil, errors.New("empty filename")
+	}
+
+	// construct the path
+	name = fmt.Sprintf("%s/%s", f.config.RemoteDir, name)
+
+	// Issue STOR command to create the empty file on FTP server
+	err := f.conn.Stor(name, emptyReader)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("Created file %s", name)
+
+	s := strings.Split(name, "/")
+
+	// Retrieve the file from FTP server and return a file handle
+	res, err := f.conn.Retr(name)
+	if err != nil {
+		return nil, err
+	}
+
+	defer res.Close()
+
+	return &ftpFile{
+		response: res,
+		name:     s[len(s)-1],
+		path:     name,
+		conn:     f.conn,
+	}, nil
+}
+
+// Mkdir creates a directory on the FTP server.
+func (f *ftpFileSystem) Mkdir(name string, _ os.FileMode) error {
+	if name == "" {
+		return errors.New("empty directory name")
+	}
+
+	name = fmt.Sprintf("%s/%s", f.config.RemoteDir, name)
+
+	return f.conn.MakeDir(name)
+}
+
+// MkdirAll creates directories recursively on the FTP server.
+func (f *ftpFileSystem) MkdirAll(path string, _ os.FileMode) error {
+	// Split path into individual directory names
+	dirs := strings.Split(path, "/")
+
+	// Start creating directories
+	currentDir := ""
+
+	for _, dir := range dirs {
+		// Ignore empty directory names (can happen if there are double slashes)
+		if dir == "" {
+			continue
+		}
+
+		// Build the full path up to this directory
+		currentDir = filepath.Join(currentDir, dir)
+
+		// Attempt to create the directory
+		err := f.conn.MakeDir(currentDir)
+		if err != nil {
+			continue // Continue creating next directory on failure
+		}
+	}
+
+	return nil
+}
+
+// Open retrieves a file from the FTP server and returns a file handle.
+func (f *ftpFileSystem) Open(name string) (datasource.File, error) {
+	if name == "" {
+		return nil, errors.New("empty filename")
+	}
+
+	name = fmt.Sprintf("%s/%s", f.config.RemoteDir, name)
+
+	// Retrieve the file from FTP server and return a file handle
+	res, err := f.conn.Retr(name)
+	if err != nil {
+		return nil, err
+	}
+
+	s := strings.Split(name, "/")
+
+	// Return the file handle
+	return &ftpFile{
+		response: res,
+		name:     s[len(s)-1],
+		path:     name,
+		conn:     f.conn,
+	}, nil
+}
+
+// OpenFile retrieves a file from the FTP server with specified flags and permissions, and returns a file handle.
+// permissions are not clear for Ftp as file commands do not accept an argument and don't store their file permissions.
+// currently, this function just calls the Open function.
+func (f *ftpFileSystem) OpenFile(name string, _ int, _ os.FileMode) (datasource.File, error) {
```

Comment:
lets Define the File interface in the ftp package itself.
datasource and ftp package will have their different declarations

As the complete point of keeping them separate is - keeping only the different features used in the build.
This increases the build size. and we have gofr imported in ftp, which should not be the case.